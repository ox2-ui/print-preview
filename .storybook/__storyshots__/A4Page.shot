exports[`landscape`] = `
<div
  className="">
  <div
    style={
      Object {
        "color": "DARKGRAY",
        "fontFamily": "Roboto Condensed, Sans Serif",
        "fontSize": "16px",
        "left": "0",
        "marginLeft": "20px",
        "marginTop": "10px",
        "position": "absolute",
        "textTransform": "capitalize",
        "top": "0",
      }
    }>
    A4 
    landscape
     preview
  </div>
  <div
    style={
      Object {
        "border": "1px dashed GAINSBORO",
        "height": "210mm",
        "margin": "40px 20px 20px 20px",
        "width": "297mm",
      }
    }>
    <div
      style={
        Object {
          "backgroundColor": "THISTLE",
          "height": "100%",
        }
      } />
  </div>
</div>
`;

exports[`portrait`] = `
<div
  className="">
  <div
    style={
      Object {
        "color": "DARKGRAY",
        "fontFamily": "Roboto Condensed, Sans Serif",
        "fontSize": "16px",
        "left": "0",
        "marginLeft": "20px",
        "marginTop": "10px",
        "position": "absolute",
        "textTransform": "capitalize",
        "top": "0",
      }
    }>
    A4 
    portrait
     preview
  </div>
  <div
    style={
      Object {
        "border": "1px dashed GAINSBORO",
        "height": "297mm",
        "margin": "40px 20px 20px 20px",
        "width": "210mm",
      }
    }>
    <div
      style={
        Object {
          "backgroundColor": "MEDIUMAQUAMARINE",
          "height": "100%",
        }
      } />
  </div>
</div>
`;
