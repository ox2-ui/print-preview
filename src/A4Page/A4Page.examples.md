### A4Page

    <A4Page/>
