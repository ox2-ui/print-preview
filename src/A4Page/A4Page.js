import React, { PropTypes } from 'react';

const styles = {
  wrapper: {
    display: 'flex',
    position: 'relative',
  },
  wrapperInner: {
    padding: '40px 20px 20px 20px',
  },
  pageWrapper: {
    border: '1px dashed GAINSBORO',
  },
  label: {
    position: 'absolute',
    top: '0',
    left: '0',
    marginTop: '12px',
    marginLeft: '22px',
    fontSize: '16px',
    color: 'DARKGRAY',
    fontFamily: 'Roboto Condensed, sans-serif',
    textTransform: 'capitalize',
  },
};

/**
 * A4Page Component
 */
const A4Page = ({ className, style, landscape, children }) => {
  const size = landscape ? { height: '210mm', width: '297mm' } : { height: '297mm', width: '210mm' };

  return (
    <div className={className} style={{...styles.wrapper, ...style}}>
      <div style={styles.label}>A4 {landscape ? 'landscape' : 'portrait'} preview</div>
      <div style={styles.wrapperInner}>
        <div style={{...styles.pageWrapper, ...size}}>
          {children}
        </div>
      </div>
    </div>
  );
};

A4Page.propTypes = {
  /**
   * The contents of the `A4Page`.
   */
  children: PropTypes.node.isRequired,
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * If set to true page renders in landscape orientation
   */
  landscape: PropTypes.bool,
  /**
   * Override inline-styles of the root element.
   */
  style: PropTypes.object,
};

export default A4Page;
