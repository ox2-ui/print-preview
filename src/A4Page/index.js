export default from './A4Page';
