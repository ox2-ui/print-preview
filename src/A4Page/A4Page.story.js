import React from 'react';
import { storiesOf } from '@kadira/storybook';
import A4Page from './A4Page';

storiesOf('A4Page', module)
  .add('landscape', () => (
    <A4Page landscape={true}>
      <div style={{backgroundColor: 'THISTLE', height: '100%'}} />
    </A4Page>
  ))
  .add('portrait', () => (
    <A4Page>
      <div style={{backgroundColor: 'MEDIUMAQUAMARINE', height: '100%'}} />
    </A4Page>
  ))
  .add('inside scrollable window', () => (
    <div style={{width: '500px', height: '700px', overflow: 'scroll', border: '2px solid grey'}}>
      <A4Page>
        <div style={{backgroundColor: 'MEDIUMAQUAMARINE', height: '100%'}} />
      </A4Page>
    </div>
  ));
