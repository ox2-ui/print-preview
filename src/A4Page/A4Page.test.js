/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import A4Page from './A4Page';

const handler = () => true;

it('renders landscape correctly', () => {
  const wrapper = mount(
    <A4Page landscape={true}>
      <div>Children</div>
    </A4Page>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders portrait correctly', () => {
  const wrapper = mount(
    <A4Page>
      <div>Children</div>
    </A4Page>,
  );

  expect(wrapper).toMatchSnapshot();
});
