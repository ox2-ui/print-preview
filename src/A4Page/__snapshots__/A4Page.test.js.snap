// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders landscape correctly 1`] = `
<A4Page
  landscape={true}
>
  <div
    style={
      Object {
        "display": "flex",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "color": "DARKGRAY",
          "fontFamily": "Roboto Condensed, sans-serif",
          "fontSize": "16px",
          "left": "0",
          "marginLeft": "22px",
          "marginTop": "12px",
          "position": "absolute",
          "textTransform": "capitalize",
          "top": "0",
        }
      }
    >
      A4 
      landscape
       preview
    </div>
    <div
      style={
        Object {
          "padding": "40px 20px 20px 20px",
        }
      }
    >
      <div
        style={
          Object {
            "border": "1px dashed GAINSBORO",
            "height": "210mm",
            "width": "297mm",
          }
        }
      >
        <div>
          Children
        </div>
      </div>
    </div>
  </div>
</A4Page>
`;

exports[`renders portrait correctly 1`] = `
<A4Page>
  <div
    style={
      Object {
        "display": "flex",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "color": "DARKGRAY",
          "fontFamily": "Roboto Condensed, sans-serif",
          "fontSize": "16px",
          "left": "0",
          "marginLeft": "22px",
          "marginTop": "12px",
          "position": "absolute",
          "textTransform": "capitalize",
          "top": "0",
        }
      }
    >
      A4 
      portrait
       preview
    </div>
    <div
      style={
        Object {
          "padding": "40px 20px 20px 20px",
        }
      }
    >
      <div
        style={
          Object {
            "border": "1px dashed GAINSBORO",
            "height": "297mm",
            "width": "210mm",
          }
        }
      >
        <div>
          Children
        </div>
      </div>
    </div>
  </div>
</A4Page>
`;
